module/zfs/vdev_raidz.c

```diff
@@ -1970,8 +1970,15 @@ vdev_raidz_open(vdev_t *vd, uint64_t *asize, uint64_t *max_asize,
 		    cvd->vdev_physical_ashift);
 	}
 
-	*asize *= vd->vdev_children;
-	*max_asize *= vd->vdev_children;
+	if (vd->vdev_rz_expanding) {
+		*asize *= vd->vdev_children - 1;
+		*max_asize *= vd->vdev_children - 1;
```

Comment:
I think this makes sense, because during an expansion, the raidz vdev's asize is still the old asize, until it completes.  If we export/import the pool (or reboot), the asize should remain set to the old asize.